appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -113,6 +113,8 @@ export class SiteNameStep extends AzureNameStep<SiteNameStepWizardContext> {
 
         if (name.length < siteNamingRules.minLength || name.length > siteNamingRules.maxLength) {
             return vscode.l10n.t('The name must be between {0} and {1} characters.', siteNamingRules.minLength, siteNamingRules.maxLength);
+        } else if ((!/^[a-z]([-a-z0-9]*[a-z0-9])?$/.test(name)) && this._siteFor === "containerizedFunctionApp") {
```

Comment:
nit: You should check the siteFor property first so the conditional can just abort if it's not a containerized app. It's probably small, but still a wasted check.